addons/arsenal/functions/fnc_attributeAddItems.sqf

```diff
@@ -68,49 +72,51 @@ if (_category == -1) exitWith {
 
 // Get list of category items
 private _categoryItems = switch (true) do {
+    // Weapons
     case (_category < 3): {
-        _configItems select 0 select _category;
+        _configItems select IDX_VIRT_WEAPONS select _category;
     };
+    // Weapon attachments
     case (_category < 7): {
-        _configItems select 1 select (_category - 3);
+        _configItems select IDX_VIRT_ATTACHMENTS select (_category - 3);
     };
+    // Other
     default {
         _configItems select (_category - 5);
     };
 };
 
 // Get config for current category
 private _config = switch (true) do {
-    case (_category in [7, 20, 21]): {
-        configFile >> "CfgMagazines";
-    };
-    case (_category == 11): {
-        configFile >> "CfgVehicles";
-    };
-    case (_category == 12): {
-        configFile >> "CfgGlasses";
-    };
-    default {
-        configFile >> "CfgWeapons";
-    };
+    case (_category in [7, 20, 21]): {_cfgMagazines};
+    case (_category == 11): {_cfgVehicles};
+    case (_category == 12): {_cfgGlasses};
+    default {_cfgWeapons};
 };
 
+private _displayName = "";
+private _picture = "";
+private _symbol = SYMBOL_ITEM_NONE;
+private _alpha = 0;
+private _index = -1;
```

Comment:
What is the reasoning for this change?

Reply:
As I see it and have tested it in the past, it helps with performance.

Instead of deleting and allocating memory space for a variable each cycle, it reuses the same memory address.
That being said I don't actually know, the previous sentence being an assumption.

Reply:
Checks out from my experience.